src/AutoBuild.jl

```diff
@@ -988,6 +997,18 @@ function build_jll_package(src_name::String, build_version::VersionNumber, code_
     mkpath(joinpath(code_dir, "src", "wrappers"))
 
     platforms = keys(build_output_meta)
+    if collect(platforms) == [AnyPlatform()]
+        # XXX: `Pkg.jl` v1.5 doesn't know about `AnyPlatform`, this makes it
+        # very hard to write a single "any" wrapper that can be easily resolved
+        # by current version of `Pkg.jl` without hacking the generated JLL
+        # package.  The easiest thing to do is to generate the wrappers for all
+        # the supported platforms, and populate `Artifacts.toml` with an entry
+        # for all platforms, all refererencing the same single tarball.  Until
+        # the far away day we'll have proper support for `AnyPlatform` also on
+        # the JLL side.
+        platforms = supported_platforms()
+        build_output_meta = Dict(p => build_output_meta[AnyPlatform()] for p in platforms)
+    end
```

Comment:
Can't we instead create an Artifact that is not platform-specific?  I don't think we need to teach Pkg about AnyPlatform at all.

Reply:
> Can't we instead create an Artifact that is not platform-specific?

This is what I tried initially, and it failed in several places, because [`select_platform`](https://github.com/JuliaLang/Pkg.jl/blob/581e70c64ba1e409c292b190cabbf99530b2142a/src/BinaryPlatforms.jl#L769-L796) and [`do_artifact""`](https://github.com/JuliaLang/Pkg.jl/blob/581e70c64ba1e409c292b190cabbf99530b2142a/src/Artifacts.jl#L1005-L1024): when they have to do with an unknown platform they go haywire.